### ISSUE TITLE
:lady_beetle: lrzip-next segfaults when STDIN + -o filename used.

### ISSUE THREAD

**Issue by pete4abw** [pete4abw/lrzip-next#125]
### lrzip-next Version

0.11.2

### lrzip-next command line

gunzip -c file.gz | lrzip-next -o file.lrz

### What happened?

Segfault

### What was expected behavior?

File compresses to file.lrz

### Steps to reproduce

pipe a file to lrzip-next

`gunzip -c file.gz  | lrzip-next -vv -o file.lrz`

### Relevant log output

```shell
$ gunzip -c file.gz  | lrzip-next -vv -o file.lrz
The following options are in effect for this COMPRESSION.
Threading is ENABLED. Number of CPUs detected: 8
Detected 16,552,353,792 bytes ram
Nice Value: 19
Show Progress
Max Verbose
Output Filename Specified: file.lrz
Temporary Directory set as: /tmp/
Compression mode is: LZMA. LZ4 Compressibility testing enabled
Compression level 7
RZIP Compression level 7
Initial LZMA Dictionary Size: 33,554,432
MD5 Hashing Used
Heuristically Computed Compression Window: 52 = 5,200MB
Storage time in seconds 1,391,590,662
Segmentation fault
```

### Please provide system details

OS Distro: Slackware 64 current
Kernel Version (uname -a): 6.3.3
System ram (free -h): lots

### Additional Context

This error does not occur if `lrzip-next` output directed to STDOUT.
This error also does not occur if the input filename is specified as `-`.

`$ gunzip -c file.gz  | lrzip-next -vv > file.lrz`
or
`$ gunzip -c file.gz  | lrzip-next -vv -o file.lrz -`

The solution may be to set the default input file as `-`  when STDIN is detected and/or when `-o` option is used.

**Comment by pete4abw** [comment]
Proposed fix in whats-next branch.

**Comment by pete4abw** [comment]
fixed in lzma-22.01
